api/src/main/java/com/lightbend/lagom/javadsl/api/broker/Subscriber.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.api.broker;
+
+import java.util.concurrent.CompletionStage;
+
+import akka.Done;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Source;
+
+/**
+ * A Subscriber for consuming messages from a message broker.
+ *
+ * @param <Message> The message type.
+ *
+ * @note This class is not meant to be extended by client code.
+ */
+public interface Subscriber<Message> {
+
+  /**
+   * Subscribers with the same group id belong to the same group.
+   * It is guaranteed that a set of subscribers with the same group id 
+   * acts as if it was a single logical subscriber, i.e., each message 
+   * is processed only by a single subscriber in the group.   
+   */
+  public static interface GroupId {
+    String groupId();
+  }
+
+  /**
+   * Thrown when an illegal group id is selected.
+   */
+  @SuppressWarnings("serial")
+  public static class IllegalGroupId extends Exception {
```

Comment:
I'll not go into that debate, but I disagree ;-)
That adds a lot of boilerplate. It will not be resolved in runtime, it's likely a programmer mistake and need to be fixed in the source of the program. I would say this is a plain `IllegalArgumentException`.